pytest_asyncio/plugin.py

```diff
@@ -88,3 +88,17 @@ def unused_tcp_port():
     with closing(socket.socket()) as sock:
         sock.bind(('127.0.0.1', 0))
         return sock.getsockname()[1]
+
+
+@pytest.fixture
+def unused_tcp_port_factory():
+    """A factory function, producing different unused TCP ports."""
+    produced = set()
+
+    def factory():
+        port = unused_tcp_port()
+        while port in produced:
+            port = unused_tcp_port
```

Comment:
Good catch! Of course the bug is in the section not tested. I'll add a test and get coverage up to 100%.